source/MaterialXShaderGen/ShaderGenerators/Glsl/TexCoordGlsl.cpp

```diff
@@ -1,46 +1,45 @@
-#include <MaterialXShaderGen/ShaderGenerators/Glsl/OgsFx/TexCoordOgsFx.h>
-#include <MaterialXShaderGen/ShaderGenerator.h>
-#include <MaterialXShaderGen/SgNode.h>
-#include <MaterialXShaderGen/HwShader.h>
+#include <MaterialXShaderGen/ShaderGenerators/Glsl/TexCoordGlsl.h>
 
 namespace MaterialX
 {
 
-SgImplementationPtr TexCoordOgsFx::creator()
+SgImplementationPtr TexCoordGlsl::creator()
 {
-    return std::make_shared<TexCoordOgsFx>();
+    return std::make_shared<TexCoordGlsl>();
 }
 
-void TexCoordOgsFx::registerInputs(const SgNode& node, ShaderGenerator& shadergen, Shader& shader)
+void TexCoordGlsl::createVariables(const SgNode& node, ShaderGenerator& shadergen, Shader& shader_)
 {
+    HwShader& shader = static_cast<HwShader&>(shader_);
+
     const SgInput* indexInput = node.getInput(INDEX);
     const string index = indexInput ? indexInput->value->getValueString() : "0";
     const string type = shadergen.getSyntax()->getTypeName(node.getOutput()->type);
 
-    shader.registerAttribute(Shader::Variable(type, "inUV" + index, "TEXCOORD" + index));
-    shader.registerVarying(Shader::Variable(type, "UV" + index, "TEXCOORD" + index));
+    shader.createAppData(DataType::VECTOR2, "i_texcoord" + index);
```

Comment:
We might want to use leave this as a vec3, or do we have a channel count we can use?

Reply:
This is a regression bug, thanks for finding it. We have the type used already above, and can just use that. Will fix it.

Reply:
Thanks.